### ISSUE TITLE
loss 0.07

### ISSUE THREAD

**Issue by Amore-HDU** [Neerajj9/CSRNet-keras#2]
Hi，when I git your project to run , I found loss is so lower that it between 0.07. When I run 200 epochs ,I tried to test ,prediction result is all 0.I want to know why? @Neerajj9

**Comment by Neerajj9** [comment]
Hello ! Please check your preprocessing. Check whether you have normalized the images using the mean and standard deviation as provided in the code ( both while training and inference ). 
Also, the loss function mentioned in the Paper, the Pytorch code as well as this repository take into consideration the MAE of the count produced from the density maps and not the MAE of the density map directly. Hope this helps !

**Comment by Amore-HDU** [comment]
Thanks for your reply, I just directly run your code,so I think I have deal with the images with right ways.

**Comment by Neerajj9** [comment]
Then please check that the loss function mentioned in the Paper, the Pytorch code as well as this repository take into consideration the MAE of the count produced from the density maps and not the MAE of the density map directly.

**Comment by macy111** [comment]
I'v got the same problem when I run the code.  How did you trained the weights provided in the program? How many epochs did you use? How can I train the network to get the similar weights as you do? @Neerajj9

**Comment by ZhengPeng7** [comment]
The same problem, could you please provide how many epochs can the model be well trained? Otherwise we can stop early when the training meets some problems. Thank you.

**Comment by Aminpdi** [comment]
dear @Neerajj9

I ran exactly your code and like other guys i have problem with constant loss and 0 prediction for all images. Then i started to find what is the problem, i checked the weights in network and all of weights are getting NAN during training.

@Neerajj9, i also implemented this model in tensorflow, but the MAE doesn't reach to 65  on partA. it reaches to 83.  can i send my code to you and check it what is the problem. i exactly used the propsed model, also i used your pre trained weights vgg16 for backend weights initialization. if it works correct we can share it here as an another implementation. 

regards 
Amin

**Comment by eain3314** [comment]
你好  解决了predict=0的问题了吗

**Comment by eain3314** [comment]
@Amore-HDU 你好 你解决这个问题了吗？ 谢谢

**Comment by chajnoven** [comment]
> Thanks for your reply, I just directly run your code,so I think I have deal with the images with right ways.

Hello, I have the same problem. So I want to ask whether you have soloved this problem. It is difficult for me to deal with it. And I run it in colab.

**Comment by mohammedayub44** [comment]
Seems like this is a re-occurring problem, the code doesn't run out of the box as provided in the iPython scripts. I ran it "as-is" for a day on GPU to get the model and it performs pretty bad on both the test sets. 
Looking to see what changes others have made to get MAE as low as mentioned. @Aminpdi

**Comment by RanRan-Margaret** [comment]
I have same problems,predict=0,I don't know why,It's really tough to me.
